console/src/components/Common/utils/v1QueryUtils.ts

```diff
@@ -98,6 +98,62 @@ export const getDropPermissionQuery = (
   };
 };
 
+export const getInsertUpQuery = (
+  tableDef: TableDefinition,
+  insertion: object
+) => {
+  return {
+    type: 'insert',
+    args: {
+      table: tableDef,
+      returning: [],
+      object: insertion,
+    },
+  };
+};
+
+type DeleteObjectWithId = {
+  id: number;
+};
+
+type PrimaryKeyInfo = {
+  table_name: string;
+  table_schema: string;
+  constraint_name: string;
+  columns: Array<string>;
+};
+
+export const getInsertDownQuery = (
+  tableDef: TableDefinition,
+  insertion: unknown,
+  primaryKeyInfo: PrimaryKeyInfo
+) => {
+  let whereClause = {};
+  const isIDPresent = (insertion as DeleteObjectWithId)?.id;
+
+  if (!isIDPresent) {
+    const primaryKeys = primaryKeyInfo.columns;
+    if (primaryKeys.length) {
+      const chosenKey = primaryKeys[0];
+      whereClause = { [chosenKey]: (insertion as any)[chosenKey] };
```

Comment:
@beerose need some help here. I have done a 😱 here with `any`

Reply:
You can provide a typing for `insertion` in line 128 as `Record<string, any>`.